### PR TITLE
ORAS Project Governance

### PR DESCRIPTION
**Pull request by SteveLasker**
Adds governance, establishing a hierarchy for overall oras project governance and owners, while enabling sub-project owners.
A huge thanks to @lachie83 for all the research and authoring of the document.

Signed-off-by: Steve Lasker <stevenlasker@hotmail.com>

### CODE REVIEW

**Review by jdolitsky — APPROVED** [review]
👍

File: governance/GOVERNANCE.md
Comment (deitch) [inline governance/GOVERNANCE.md:21]: Can we change this from "never" to "should not"? Rigid language can bind us in unforeseen ways.

File: governance/GOVERNANCE.md
Comment (deitch) [inline governance/GOVERNANCE.md:42]: Can we change this to "should employ"? I don't want us to be in a situation where one person leaves right after someone else from one company joins, and we suddenly are in violation, just because one company's employees stepped up for a bit. The rules should nudge us in the right direction, not suddenly find us in violation despite (or because of) extra good will.

File: governance/GOVERNANCE.md
Comment (deitch) [inline governance/GOVERNANCE.md:83]: Over how long a period? If I submit a proposal on Monday 0900 UTC, is it approved if no one objects by 0900 Tuesday? 0901 Monday (I hope not)? 0900 Monday a week later? We should have an explicit "time to object" if silence equals assent.

Comment (deitch) [reply governance/GOVERNANCE.md:83]: Do we want acknowledgment of receipt, and do we need some quorum of acknowledgement for silence=assent to kick in (or voting)? How do we handle people being on vacation?

**Review by deitch — CHANGES_REQUESTED** [review]
Some suggestions to soften language, and a suggestion around putting timeframes around voting and consensus procedures.

File: governance/GOVERNANCE.md
Comment (SteveLasker) [reply governance/GOVERNANCE.md:21]: I struggled for how to reword this, as it sounded like a double negative.
> - To never take any action while hesitant that it is the right action to take

- Should not take any action while hesitant that it is the right action to take

Sounds like they shouldn't take any action?

@lachie83, can you help with some wording? I'll do another PR to amend this if we can find better language.

File: governance/GOVERNANCE.md
Comment (SteveLasker) [reply governance/GOVERNANCE.md:42]: @lachie83 and I talked about this a bit. The concern isn't that a company would be in violation if folks leave and gaps need to be filled. It's more that this creates an action item to resolve. If it's worded as _should employ_ it leaves it open to interpretation where a company can say SHOULD isn't MUST, so they would have leeway.

File: governance/GOVERNANCE.md
Comment (SteveLasker) [reply governance/GOVERNANCE.md:83]: I'm struggling on this one as it could be interpreted two ways. If we say we must have 3 business days for folks to object, does that mean that every PR either must have unanimous consent to approve that day, or every PR must wait 3 business days before merging because we must give time for the time clock to run out, or aggree?

File: governance/GOVERNANCE.md
Comment (deitch) [reply governance/GOVERNANCE.md:21]: "Should only take action when reasonably confident that it is the right action to take"?

File: governance/GOVERNANCE.md
Comment (deitch) [reply governance/GOVERNANCE.md:42]: Fair enough. It is weighing the balance of "creating action items" vs "being in violation".

File: governance/GOVERNANCE.md
Comment (deitch) [reply governance/GOVERNANCE.md:83]: I think we need to distinguish between PRs - code changes - and major decisions. There are fundamental changes such as architecture or major new structures that should go through greater approval 9lazy consensus, give people time to object or agree), and there are the vast majority of PRs which really just takes one maintainer to sign off (fix a bug, add an option, add a new function, etc.).

If we distinguish between those, it becomes easier.

File: governance/GOVERNANCE.md
Comment (SteveLasker) [reply governance/GOVERNANCE.md:21]: I've tweaked it a minor bit, to be consistent with the other wording:
To take action when reasonably confident that it is the right action to take

File: governance/GOVERNANCE.md
Comment (SteveLasker) [reply governance/GOVERNANCE.md:83]: I think this goes back to your questions about how much detail we need to provide, being overly prescriptive, to the point of legal mumbo-jumbo, and enough clarity for the right people to do the right things, the humans to make mistakes and let them be fixed, and enough to assure the few bad don't do bad things. (or allow bad things to be undone).

Comment (deitch) [reply governance/GOVERNANCE.md:83]: > legal mumbo-jumbo

Yeah, I spend too much time in contracts; that's what I get for deciding to run my own business. 😆 

OK, let's have this "good enough" for now and improve later.

**Review by deitch — APPROVED** [review]

**Pull request merged**
